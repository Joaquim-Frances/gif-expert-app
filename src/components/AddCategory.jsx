import React, { useState } from 'react';
import PropTypes from "prop-types";




export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleImputChange = (event) => {
        setInputValue(event.target.value);

        console.log('HandleInputChange Llamado!')
    }

    const handleSubmit = (event) =>{
        event.preventDefault();

        console.log('HandelSubmit!', inputValue );

        if( inputValue.trim().length > 2 ){ 
            setCategories( categ =>[inputValue, ...categ]);
            setInputValue('');
        }
    }

    return (
        <form className="addCategory animate__animated animate__zoomIn" onSubmit={ handleSubmit }>
            <h3>Buscar categoria: </h3>
            <p> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                placeholder="Â¿Sobre que tema quieres que busque GIFs?"
                onChange= { handleImputChange }
                className="input-category"
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
