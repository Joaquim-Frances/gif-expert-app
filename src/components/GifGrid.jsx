import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'


export const GifGrid = ({category}) => {
    
   
    const { data, loading } = useFetchGifs( category );
    

    // useEffect ( ()=> {
    //     getGif( category ).then( setImages );
    // }, [ category]);
    
    return (
        <Fragment>
        <h3> { category } </h3>
        {loading && <p> Cargando datos...</p>}
        <div className='card-grid'>
                {
                    data.map( img =>{
                        return(
                        <GifGridItem 
                        key={ img.id }
                        { ...img } />
                        )
                    })
                }
        </div>
        </Fragment>
    )
}
