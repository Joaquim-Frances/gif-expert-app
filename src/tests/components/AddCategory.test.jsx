import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import '@testing-library/jest-dom';

describe('Testing all "AddCategory" functions', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    })

    test('should mantain the structure', () => {
        expect(wrapper).toMatchSnapshot();
    })
        
    test('should change after a user enters data in the input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Quim';
        input.simulate('change', {target:{ value }});

        expect(wrapper.find('p').text().trim()).toBe(value);

    })

    test('should not enter data if input value is not correct o empty', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();
        
    })

    test( 'should call setCategories and clean input box', () => {

        const input = wrapper.find('input');
        const value = 'Hola Quim';
        input.simulate('change', {target:{ value }});
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).toHaveBeenCalled();
        expect( input.prop('value')).toBe('');

    })
    
    
        

    
    
})
