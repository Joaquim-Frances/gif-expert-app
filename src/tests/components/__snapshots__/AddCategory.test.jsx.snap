// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing all "AddCategory" functions should mantain the structure 1`] = `
<form
  className="addCategory animate__animated animate__zoomIn"
  onSubmit={[Function]}
>
  <h3>
    Buscar categoria: 
  </h3>
  <p>
     
     
  </p>
  <input
    className="input-category"
    onChange={[Function]}
    placeholder="Â¿Sobre que tema quieres que busque GIFs?"
    type="text"
    value=""
  />
</form>
`;
